src/System Application/App/Azure Storage Services Authorization/src/ReadySAS/StorServAuthReadySAS.Codeunit.al

```diff
@@ -29,28 +29,36 @@ codeunit 9088 "Stor. Serv. Auth. Ready SAS" implements "Storage Service Authoriz
 
         if QueryText <> '' then
             QueryText += '&';
-        QueryText += GetSharedAccessSignature();
+        QueryText += GetSharedAccessSignature().Unwrap();
         UriBuilder.SetQuery(QueryText);
 
         UriBuilder.GetUri(Uri);
 
         HttpRequestMessage.SetRequestUri(Uri.GetAbsoluteUri());
     end;
 
-    [NonDebuggable]
-    procedure GetSharedAccessSignature(): Text
+    procedure GetSharedAccessSignature(): SecretText
     begin
         exit(SharedAccessSignature);
     end;
 
     [NonDebuggable]
     procedure SetSharedAccessSignature(NewSharedAccessSignature: Text)
     begin
-        SharedAccessSignature := NewSharedAccessSignature;
-        if SharedAccessSignature.StartsWith('?') then
-            SharedAccessSignature := DelChr(SharedAccessSignature, '<', '?');
+        SetSharedAccessSignature(SecretStrSubstNo(NewSharedAccessSignature));
```

Comment:
Shouldn't require `SecretStrSubstNo`, `Text` can go directly into `SecretText` but not vice versa.